strutil/range.go

```diff
@@ -0,0 +1,118 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2025 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package strutil
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// RangeSpan represents a span of numbers inside a range. A span with
+// equal Start and End describes a span of a single number.
```

Comment:
A mathematical span `(0, 0]` (inclusive end) has different semantics than a Go slice where `s[0:0]` returns the empty slice for any `s`. Do you think it is possible to make ranges "just like slices" so that we can have fewer surprises?